packages/web-components/@sfx/product/src/product.ts

```diff
@@ -0,0 +1,104 @@
+import { customElement, property, html, TemplateResult } from 'lit-element';
+
+import { Base } from '@sfx/base';
+
+@customElement('sfx-product')
+export default class Product extends Base {
+  @property({ type: Object }) product: ProductModel = {};
+
+  constructor() {
+    super();
+    this.updateVariant = this.updateVariant.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEventListener('sfx::product_variant_change', this.updateVariant);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    this.removeEventListener('sfx::product_variant_change', this.updateVariant);
+  }
+
+  updateVariant(e: CustomEvent) {
+    const product = e.detail;
+
+    this.product = {
+      ...this.product,
+      ...product
+    };
+  }
+
+  urlWrap(url: string, children: TemplateResult) {
+    return url ? html`<a href="${ url }">${children}</a>` : children;
+  }
+
+  additionalInfo() {
+    const { product } = this;
+
+    const properties = {
+      title: '',
+      price: 0,
+      variants: { type: 'text', items: [] },
+      productUrl: '',
+      imageAlt: '',
+      imageSrc: ''
+    } as ProductModel;
+
+    return Object.keys(product).filter(p => !(p in properties)).map(p => html`
+      <span class="sfx-product-${ p }">${ product[p] }</span>
+    `);
+  }
+
+  render() {
+    const { title, price, variants, productUrl, imageSrc, imageAlt } = this.product;
+
+    return html`
+      <style>.product-variants { padding: 0 }</style>
+      <slot name="image">
+        ${ imageSrc ? html`<img src="${ imageSrc }" alt="${ imageAlt }" />`: '' }
+      </slot>
+      <slot name="variants">
+        <ul class="product-variants">
```

Comment:
Should this be more specific to SF-X?

Reply:
@geoffcoutts Leaving this until we decide on how elements are going to be targeted via `SFX-246`